omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,297 @@
+		
+        Object savedIdentifierSource = null;
+        String identifierSourceType = validateInput(postBody.get("sourceType"));
+        ArrayList<String> errors = new ArrayList<String>();
+        String identifierTypeUuid = validateInput(postBody.get("identifierType"));
+        PatientIdentifierType patientIdentifierType = Context.getPatientService().getPatientIdentifierTypeByUuid(identifierTypeUuid);
+        String name = validateInput(postBody.get("name"));
+        String description = postBody.get("description").toString();
+
+        if (StringUtils.isBlank(identifierTypeUuid) || patientIdentifierType == null) {
+            errors.add("patient identifier type"); 
+        }
+        if (StringUtils.isBlank(name)) {
+            errors.add("name"); 
+        }
+        if (StringUtils.isBlank(description)) {
+            errors.add("description"); 
+        }
+
+        if(identifierSourceType.equals("SequentialIdentifierGenerator")) {
+
+            SequentialIdentifierGenerator identifierSource = new SequentialIdentifierGenerator();
+            String firstIdentifierBase = validateInput(postBody.get("firstIdentifierBase"));
+            String baseCharacterSet = validateInput(postBody.get("baseCharacterSet"));
+            String prefix = validateInput(postBody.get("prefix"));
+            String suffix = validateInput(postBody.get("suffix"));
+            String minLength = validateInput(postBody.get("minLength"));
+            String maxLength = validateInput(postBody.get("maxLength"));
+
+            if (StringUtils.isBlank(firstIdentifierBase)) { 
+                errors.add("first identifier base"); 
+            }
+            if (StringUtils.isBlank(baseCharacterSet)) { 
+                errors.add("base character set"); 
+            }
+            if (StringUtils.isBlank(baseCharacterSet)) { 
+                errors.add("base character set"); 
+            }
+            if (!StringUtils.isBlank(prefix)) { 
+                identifierSource.setPrefix(prefix); 
+            }
+            if (!StringUtils.isBlank(suffix)) { 
+                identifierSource.setSuffix(suffix); 
+            }
+            if (!StringUtils.isBlank(minLength)) {
+                if(StringUtils.isNumeric(minLength)) {
+                        identifierSource.setMinLength(Integer.parseInt(minLength));
+                }else {
+                        errors.add("min length");
+                }
+            }
+            if (!StringUtils.isBlank(maxLength)) {
+                if(StringUtils.isNumeric(maxLength)) {
+                        identifierSource.setMaxLength(Integer.parseInt(maxLength));
+                }else {
+                        errors.add("max length");
+                }
+            }
+            if(errors.toString().length() > 2) {
+                throw new ConversionException("The values of the following inputs are missing or invalid: " + errors.toString());
```

Comment:
Does this mean you are not throwing exceptions if you have one or two errors?

Reply:
Secondly for some of these, ConversionException is not appropriate. For instance, if one does not supply a required value like name, why would they receive a ConversionException?

Reply:
I am appending to an array list the values of any error. When i do: `errors.toString()` the output comes with braces : `["name", "description"]` and the errors inside separated by a comma. So to see if there are any errors i do `if(errors.toString().length() > 2)` so as to not count the opening and closing braces in the output string.

Reply:
Is a [ValidationException](http://docs.openmrs.org/doc/org/openmrs/api/ValidationException.html) appropriate?

Reply:
Where does ["name", "description"] come from if they are not errors?

Reply:
fixed